### PR TITLE
Healthkit service

### PR DESCRIPTION
**Pull request by filletofish**

### CODE REVIEW

File: K Group Challenge/Health Services/HealthKitService.swift [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+import Foundation
+import HealthKit
+
+struct HealthKitStaticData {
+    let age: Int
+    let gender: Sex
+}
+
+enum Sex {
+    case female, male, other
```
Comment (ostrenkiy) [inline K Group Challenge/Health Services/HealthKitService.swift:18]: Too offensive

File: K Group Challenge/Health Services/HealthKitService.swift [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+    }
+}
+
+enum HealthKitServiceError: Error {
+    case notAvailableOnDevice
+    case dataTypeNotAvailable
+}
+
+class HealthKitService {
+
```
Comment (ostrenkiy) [inline K Group Challenge/Health Services/HealthKitService.swift:45]: Empty lines

File: K Group Challenge/Health Services/HealthKitService.swift [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    func authorize(completion: @escaping(Bool, Error?) -> ()) {
+        //1. Check to see if HealthKit Is Available on this device
+        guard HKHealthStore.isHealthDataAvailable() else {
+            completion(false, HealthKitServiceError.notAvailableOnDevice)
+            return
+        }
+
+        //2. Prepare the data types that will interact with HealthKit
+        guard   let dateOfBirth = HKObjectType.characteristicType(forIdentifier: .dateOfBirth),
```
Comment (ostrenkiy) [inline K Group Challenge/Health Services/HealthKitService.swift:57]: please add linter

File: K Group Challenge/Health Services/HealthKitService.swift [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+            let biologicalSex = HKObjectType.characteristicType(forIdentifier: .biologicalSex),
```
Comment (ostrenkiy) [inline K Group Challenge/Health Services/HealthKitService.swift:58]: where is psychological

File: K Group Challenge/Health Services/HealthKitService.swift [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+            let fatPercentage = HKObjectType.quantityType(forIdentifier: .bodyFatPercentage),
+            let activity = HKObjectType.quantityType(forIdentifier: .appleExerciseTime),
+            let basalEnergy = HKObjectType.quantityType(forIdentifier: .basalEnergyBurned),
+            let activeEnergy = HKObjectType.quantityType(forIdentifier: .activeEnergyBurned) else {
+
+                completion(false, HealthKitServiceError.dataTypeNotAvailable)
+                return
+        }
+
+//        //3. Prepare a list of types you want HealthKit to read and write
```
Comment (ostrenkiy) [inline K Group Challenge/Health Services/HealthKitService.swift:71]: We have git, don't need comments

File: K Group Challenge/Health Services/HealthKitService.swift [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+
+    /// Return Kilocalories for resting energy
+    ///
+    func getActiveEnergy(completion: @escaping(Double?, Error?) -> ()) {
+        guard let activeEnergySampleType = HKSampleType.quantityType(forIdentifier: .activeEnergyBurned) else {
+            print("Mass Sample Type is no longer available in HealthKit")
+            return
+        }
+
+        getStatisticsMonth(for: activeEnergySampleType) { (statistics, numberOfDays, error) in
```
Comment (ostrenkiy) [inline K Group Challenge/Health Services/HealthKitService.swift:203]: long line

**Review by ostrenkiy — CHANGES_REQUESTED** [review]

**Comment by filletofish** [comment]
@Ostrenkiy Very nice comments! Thank you 👍 Nice catch! Will try to fix them ASAP!!

**Pull request merged**